server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -424,13 +424,24 @@ public Symbol visitMatchPredicate(MatchPredicate matchPredicate, Context ctx) {
 
         @Override
         public Symbol visitReference(Reference ref, Context ctx) {
-            if (ctx.comparisons.containsKey(ref.column())) {
+            var comparison = ctx.comparisons.get(ref.column());
+            if (comparison != null) {
+                // A boolean PK column must be treated as col = true and add another one comparison.
+                // Otherwise, we might not create a DocKey with value TRUE if it's not added in other parts of the expression.
+                if (ref.valueType().equals(DataTypes.BOOLEAN)) {
+                    // Duplicate entries are eliminated in comparison.add() method.
+                    ctx.proxyBelow = true;
+                    return comparison.add(
+                        new Function(EqOperator.SIGNATURE, List.of(ref, Literal.BOOLEAN_TRUE), EqOperator.RETURN_TYPE)
+                    );
```

Comment:
this could add col = true for a non-standalone column, ie for where bool_pk = false we would anyway add  true proxy but this is only protected by special handling/short-circuit in the `visitFunction`